misk-redis/src/main/kotlin/misk/redis/Redis.kt

```diff
@@ -517,6 +517,131 @@ interface Redis {
     member: String
   ) : Double?
 
+  /**
+   * Returns the specified range of elements in the sorted set stored at [key].
+   *
+   * ZRANGE can perform different [type]s of range queries: by index (rank), by the score, or by
+   * lexicographical order. Currently only index and score type range queries are supported.
+   * See [ZRangeType] for different types of range queries.
+   *
+   * You can specify the [start] and [stop] of the range you want to filter by.
+   * Depending on the [type] you will have to use the appropriate type of [ZRangeMarker].
+   *
+   * The order of elements is from the lowest to the highest score.
+   * Elements with the same score are ordered lexicographically.
+   *
+   * Setting [reverse] reverses the ordering, so elements are ordered from highest to lowest score,
+   * and score ties are resolved by reverse lexicographical ordering.
+   *
+   * The [limit] argument can be used to obtain a sub-range from the matching elements.
+   * See [ZRangeLimit] for more info.
+   */
+  fun zrange(
+    key: String,
+    type: ZRangeType = ZRangeType.INDEX,
+    start: ZRangeMarker,
+    stop: ZRangeMarker,
+    reverse: Boolean = false,
+    limit: ZRangeLimit? = null,
```

Comment:
Can we simply re-use the existing params class in Jedis? I worry that this reimplements a bunch of Jedis logic. This API should be very thin wrapper around Jedis
```suggestion
    params: ZRangeParams,
```

Reply:
> Can we simply re-use the existing params class in Jedis?

ZRangeParams does not support use cases like include/exclude in ranges, infinite ranges, limits. For that, jedis also uses separate methods altogether.

Reply:
Do we need to support all of those? Which form will we use in Shadow?

It looks like it's possible to add a limit to `ZRangeParam`. Does that not work? Want to pair on this?
![Screenshot 2024-01-09 at 12 17 21 PM](https://github.com/cashapp/misk/assets/588701/edc68851-b2cf-48b8-b76c-415ad2a7e536)

Reply:
maybe You can use a sealed class like this? 
```
sealed interface ZRangeParams 
data class ZRangeByScore(...) : ZRangeParams
data class ZRange(...) : ZRangeParams
```